plugins/by-name/treesitter/default.nix

```diff
@@ -365,46 +233,110 @@ lib.nixvim.plugins.mkNeovimPlugin {
+
+        if has_configs_module then
+          -- Master branch: Use the traditional configs API
+          require('nvim-treesitter.configs').setup(${lib.nixvim.toLuaObject cfg.settings})
+        else
+          -- Main branch: Pass install_dir to setup() and enable features via autocommands
+      ''
+      + (optionalString (mainBranchSettings != { }) ''
+        require'nvim-treesitter'.setup(${lib.nixvim.toLuaObject mainBranchSettings})
+      '')
+      + ''
+
+        -- Enable features via autocommands for main branch
+      ''
+      + (optionalString highlightEnabled ''
+        vim.api.nvim_create_autocmd('FileType', {
+          pattern = '*',
+          callback = function()
+            pcall(vim.treesitter.start)
+          end,
+        })
+      '')
+      + (optionalString indentEnabled ''
+        vim.api.nvim_create_autocmd('FileType', {
+          pattern = '*',
+          callback = function()
+            vim.bo.indentexpr = "v:lua.require'nvim-treesitter'.indentexpr()"
+          end,
+        })
       '')
       + ''
-        require('nvim-treesitter.configs').setup(${lib.nixvim.toLuaObject cfg.settings})
+        end
       ''
       + (optionalString (cfg.languageRegister != { }) ''
+
         do
           local __parserFiletypeMappings = ${lib.nixvim.toLuaObject cfg.languageRegister}
 
           for parser_name, ft in pairs(__parserFiletypeMappings) do
-            require('vim.treesitter.language').register(parser_name, ft)
+            vim.treesitter.language.register(parser_name, ft)
           end
         end
       '');
 
-    extraFiles = mkIf cfg.nixvimInjections { "queries/nix/injections.scm".source = ./injections.scm; };
+      extraFiles = mkIf cfg.nixvimInjections { "queries/nix/injections.scm".source = ./injections.scm; };
 
-    # Install the grammar packages if enabled
-    plugins.treesitter.packageDecorator = lib.mkIf cfg.nixGrammars (
-      pkg: pkg.withPlugins (_: cfg.grammarPackages)
-    );
+      # Install the grammar packages if enabled
+      plugins.treesitter.packageDecorator = lib.mkIf cfg.nixGrammars (
+        pkg: pkg.withPlugins (_: cfg.grammarPackages)
+      );
 
-    warnings = lib.nixvim.mkWarnings "plugins.treesitter" (
-      lib.map (packageName: {
-        when = !cfg.nixGrammars && !config.dependencies.${packageName}.enable;
-        message = ''
-          `${packageName}` is required to build grammars as you are not using `${opt.nixGrammars}`.
-          You may want to set `${options.dependencies.${packageName}.enable}` to `true`.
+      autoGroups.nixvim_treesitter.clear = true;
+
+      autoCmd = lib.optional cfg.folding.enable {
+        event = "FileType";
+        group = "nixvim_treesitter";
+        pattern = "*";
+        callback.__raw = ''
+          function()
+            vim.wo[0][0].foldexpr = 'v:lua.vim.treesitter.foldexpr()'
+            vim.wo[0][0].foldmethod = 'expr'
+          end
         '';
```

Comment:
Just noticed this, why manually create auto commands for highlight and indent but not for folding?

Reply:
makes sense